.pre-commit-config.yaml

```diff
@@ -35,3 +35,11 @@ repos:
     hooks:
       - id: ruff
       - id: ruff-format
+
+  - repo: https://github.com/pre-commit/mirrors-mypy
+    rev: v1.9.0
+    hooks:
+      - id: mypy
+        exclude: ^(tests/)
+        include: clouds/azure/
```

Comment:
Following @myakove [comment](https://github.com/RedHatQE/cloud-tools/pull/475#discussion_r1564564503), should we remove or modify the hook ?

Reply:
it was an example for the hook; should be updated according to the repo

Reply:
I removed tests exclude from the hook, all `clouds/azure/ ` modules functions use typing.

Reply:
can you add `args: [--ignore-missing-imports, --disallow-untyped-defs]` and check that pre-commit still pass?

Reply:
It gives multiple missing type annotation errors for aws/gcp/cli modules.
I think it will be better to add this hook in a separate PR and add missing typing, wyt?